### PR TITLE
fix schema cache

### PR DESCRIPTION
**Pull request by tanjabrzak**

### CODE REVIEW

**Comment by kastner** [comment]
I looked into this yesterday with @tanjabrzak and here's the deal...

When an XHR request is made to populate the column data, the `ColumnsController` handles it.
This line: https://github.com/lumoslabs/aleph/blob/master/app/controllers/columns_controller.rb#L7 accesses a Singleton that is a cache on top of the Redis cache. When that cache is empty, `Schemas::Descriptors` (notice the `s`) will instanciate `Schemas::Descriptor` (no `s`) here: https://github.com/lumoslabs/aleph/blob/f3e7fc8a/lib/schemas/descriptors.rb#L7

Looking at the [initalizer](https://github.com/lumoslabs/aleph/blob/f3e7fc8a/lib/schemas/descriptor.rb#L25-L30) for that class, we see a new thread is started, and [schema_refresher](https://github.com/lumoslabs/aleph/blob/f3e7fc8a/lib/schemas/descriptor.rb#L61) is called. 
That starts an infinite loop that calls [refresh_schema](https://github.com/lumoslabs/aleph/blob/f3e7fc8a/lib/schemas/descriptor.rb#L68) then sleeps for `SCHEMA_REFRESH_INTERVAL` seconds (one day by default).
`refresh_schema` will get the schema, and then put it in redis, with an [EXPIRE](https://github.com/lumoslabs/aleph/blob/f3e7fc8a/lib/schemas/redis_store.rb#L19:L19) of 3 days.

TL;DR
Each Puma thread will fetch schema on first request and store it again in redis, while also keeping a thread-local copy for 1 day.
A better strategy would be to not have the extra thread-local cache and use a shared cache (like the one that's already in redis). That's what @tanjabrzak PR does!
I think a good next step would be to remove the Singleton altogether, but it isn't necessary.
